api/v1alpha3/machinedeployment_webhook.go

```diff
@@ -129,4 +137,11 @@ func PopulateDefaultsMachineDeployment(d *MachineDeployment) {
 			d.Spec.Strategy.RollingUpdate.MaxUnavailable = &ios0
 		}
 	}
+
+	// If no selector has been provided, add label and selector for the
+	// MachineDeployment's name as a default way of providing uniqueness.
+	if len(d.Spec.Selector.MatchLabels) == 0 && len(d.Spec.Selector.MatchExpressions) == 0 {
+		d.Spec.Selector.MatchLabels[MachineDeploymentLabelName] = d.Name
+		d.Spec.Template.Labels[MachineDeploymentLabelName] = d.Name
+	}
```

Comment:
Might be good to go ahead and add a test for this in `machinedeployment_webhook.go`

Reply:
Agreed, will add.

Reply:
added